lib/SILOptimizer/Utils/CanonicalizeInstruction.cpp

```diff
@@ -294,14 +297,25 @@ splitAggregateLoad(LoadOperation loadInst, CanonicalizeInstruction &pass) {
     }
     pass.notifyNewInstruction(**lastNewLoad);
 
+    // If the debug_values for the original load will be deleted, then recover
+    // as much debug info as possible by creating debug_value fragments for each
+    // new partial load.
+    //
+    // FIXME: This drops debug info at -Onone because debug fragments are only
+    // supported at -O, and load-splitting is required at -Onone for exclusivity
+    // diagnostics. One possible fix would be to opt-out of shadow stack
+    // variable in this case and use debug fragments instead.
```

Comment:
I think [this line](https://github.com/apple/swift/blob/17f1cf92973d2c1b3d5fe6b70f4a33771a303e3a/lib/IRGen/IRGenDebugInfo.cpp#L2469) is blocking us from using fragment in -Onone. Though I don't know why people adding it in the first place.
But anyway, I think using debug fragment instead of shadow stack slots is a neat solution.